src/serialize.h

```diff
@@ -571,7 +571,7 @@ void Serialize_impl(Stream& os, const prevector<N, T>& v, const unsigned char&)
 {
     WriteCompactSize(os, v.size());
     if (!v.empty())
-        os.write((char*)&v[0], v.size() * sizeof(T));
+        os.write((char*) v.data(), v.size() * sizeof(T));
```

Comment:
Yeah that seems obviously incorrect, but I _think_ I understand why it was done (or at least have a theory). Because the methods are all generic w.r.t. the stream type, they probably are made to work with some arbitrary stream that can't accept a `const char *`. Hopefully not the case anymore...

Anyways, probably best to fix those up separately as you suggest.